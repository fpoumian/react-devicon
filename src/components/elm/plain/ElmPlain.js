import React from 'react'
import PropTypes from 'prop-types'
import SVGDeviconInline from '../../_base/SVGDeviconInline'

import iconSVG from './ElmPlain.svg'

/** ElmPlain */
function ElmPlain({ width, height, className }) {
  return (
    <SVGDeviconInline
      className={'ElmPlain' + ' ' + className}
      iconSVG={iconSVG}
      width={width}
      height={height}
    />
  )
}

ElmPlain.propTypes = {
  className: PropTypes.string,
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
}

export default ElmPlain
