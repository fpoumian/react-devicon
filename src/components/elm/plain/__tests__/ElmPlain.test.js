import React from 'react'
import renderer from 'react-test-renderer'
import ElmPlain from '../index.js'

it('renders correctly', () => {
  const tree = renderer.create(<ElmPlain />).toJSON()
  expect(tree).toMatchSnapshot()
})
