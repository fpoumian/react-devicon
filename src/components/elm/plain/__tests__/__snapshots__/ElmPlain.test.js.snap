// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="SVGInline ElmPlain undefined"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg class=\\"SVGInline-devicon ElmPlain-devicon undefined-devicon\\" id=\\"Elm\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 128 128\\"><defs><style>.cls-1{fill:#34495e}</style></defs><title>Artboard 14</title><g id=\\"plain\\"><path class=\\"cls-1\\" d=\\"M64 60.74l25.65-25.65h-51.3L64 60.74zM7.91 4.65l25.83 25.84h56.17L64.07 4.65H7.91z\\"/><path class=\\"cls-1\\" transform=\\"rotate(-45 95.27 63.928)\\" d=\\"M75.42 44.17h39.71V83.7H75.42z\\"/><path class=\\"cls-1\\" d=\\"M123.35 57.42V4.65H70.58l52.77 52.77zM60.74 64L4.65 7.91V120.1L60.74 64zM98.47 95.21l24.88 24.89V70.33L98.47 95.21zM64 67.26L7.91 123.35h112.18L64 67.26z\\"/></g></svg>",
    }
  }
/>
`;
